nitro-protocol/contracts/InterestBearingApp.sol

```diff
@@ -0,0 +1,150 @@
+pragma solidity 0.8.17;
+pragma experimental ABIEncoderV2;
+
+import './interfaces/IForceMoveApp.sol';
+import './libraries/NitroUtils.sol';
+import {ExitFormat as Outcome} from '@statechannels/exit-format/contracts/ExitFormat.sol';
+
+// InterestBearingApp is a ForceMoveApp that allows a lender to earn interest
+// on a deposit. It functions as a ConsensusApp with the following additional rule:
+//   - the lender can unilaterally transition from state n to state n+1,
+//     forcing calculated interest into the lender's Outcome allocation
+contract InterestBearingApp is IForceMoveApp {
+    struct Funds {
+        address[] asset;
+        uint256[] amount;
+    }
+
+    struct InterestAppData {
+        // the per-block interest rate, expressed as the denominator of a fraction
+        // eg, 1% per block would be expressed by interestPerBlockDivisor = 100,
+        //     2% per block would be expressed by interestPerBlockDivisor = 50,
+        //     0.1% per block would be expressed by interestPerBlockDivisor = 1000,
+        // etc.
+        uint256 interestPerBlockDivisor;
+        // the block number of the latest principal adjustment
+        uint256 blocknumber;
+        // the current principal. Decreases as the borrower earns via the channel.
+        Funds principal;
+        // the total interest collected so far. Strictly increasing.
+        // The value of the lender's allocation can never be less than this:
+        // ie, when the lender's collectedInterest grows to be equal to their
+        // allocation, the channel is effectively spent and can be concluded.
+        Funds collectedInterest;
+    }
+
+    enum AllocationIndicies {
+        borrower, // intends to earn service fees up to a limit of the financier's deposit
```

Comment:
```suggestion
        borrower, // intends to earn service fees up to a limit of the lender's deposit
```
Should we use the term `lender` since that's what we use for the enum below?